### PR TITLE
Add CUP Terrains Compatibility

### PR DESCRIPTION
**Pull request by Mike-MF**
**When merged this pull request will:**
- Merges CUP Terrains Compat from CUP to ACE.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by PabstMirror** [comment]
`FuelStation: House_Small_F`
is some kind of tank/pump combo
configs are a little trickey because we'd want to move main actions to the pump and I'm not sure we can do that
I've never seen a map actually use this object so it's probably something we can ignore

**Comment by Mike-MF** [comment]
`Land_Benzina_schnell` is used on Sahrani, The interaction point to start refueling is between both pumps but the actual fuel hose and nozzle come from one of the 2 available pumps.

**Review by PabstMirror — APPROVED** [review]

File: addons/compat_cup_terrains/config.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        units[] = {};
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {
+            "CABuildings",
+            "CAStructuresHouse_A_FuelStation",
+            "CAStructures_E_Ind_Ind_FuelStation",
+            "CAStructures_PMC_FuelStation",
+            "CUP_Buildings_Config",
+            "ace_refuel"
```
Comment (jonpas) [inline addons/compat_cup_terrains/config.cpp:16]:
```suggestion
            "ace_refuel" // not a sub-component because it's all this compat does
```

**Review by jonpas — APPROVED** [review]

**Comment by acknowledge14** [comment]
Hi, Could you please also do something to CUP CWA's gas stations and vehicle maintain depots?

**Comment by Drofseh** [comment]
> Hi, Could you please also do something to CUP CWA's gas stations and vehicle maintain depots?

CWA is not part of CUP and is developed by a different team. It would need a seperate compat.

**Review by PabstMirror — APPROVED** [review]

**Review by LinkIsGrim — APPROVED** [review]

**Pull request merged**